src/database/transaction_id.cpp

```diff
@@ -12,11 +12,19 @@ TransactionId::TransactionId() {
   id_value = counter.fetch_add(1);
 }
 
+TransactionId::TransactionId(long id) {
+  this->id_value = id;
+}
+
 long TransactionId::get_id_value() {
   return id_value;
 }
 
 bool TransactionId::operator==(TransactionId & other) {
   return id_value == other.id_value; 
 }
+
+int HashCode() {
```

Comment:
The `hashcode` function is a Java-specific thing, Java stipulates that the hashcode function should be overriden whenever the equals function is overriden. So we don't really need it in C++, please remove it! Thanks for doing the hard work here again, though!